### PR TITLE
Updating all dependencies

### PR DESCRIPTION
**Pull request by BillFarber**
Still need to update DH - that will be a later PR.
I'm going to move Confluent Platform to a Docker cluster first.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,20 +31,22 @@ configurations {
...
-  compileOnly "org.slf4j:slf4j-api:1.7.36"
+  compileOnly "org.slf4j:slf4j-api:2.0.13"
 
   implementation 'com.marklogic:ml-javaclient-util:4.7.0'
   // Force DHF to use the latest version of ml-app-deployer, which minimizes security vulnerabilities
-  implementation "com.marklogic:ml-app-deployer:4.6.0"
+  implementation "com.marklogic:ml-app-deployer:4.7.0"
 
-  implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-csv:2.15.2"
+  implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-csv:2.17.1"
```
Comment (rjrudin) [inline build.gradle:49]: The Java Client is only on 2.15.3 - stick with that, particularly as I believe Kafka provides an isolated class loader for a plugin so we won't be impacted by whatever Jackson version Kafka is using (not certain about that, but think it's true).

File: build.gradle [hunk 2]
```diff
@@ -63,13 +65,13 @@ dependencies {
...
-  testImplementation 'net.mguenther.kafka:kafka-junit:3.5.1'
+  testImplementation 'net.mguenther.kafka:kafka-junit:3.6.0'
 
-  testImplementation "org.apache.avro:avro-compiler:1.11.1"
+  testImplementation "org.apache.avro:avro-compiler:1.11.3"
 
   // Forcing logback to be used for test logging
-  testImplementation "ch.qos.logback:logback-classic:1.3.5"
-  testImplementation "org.slf4j:jcl-over-slf4j:1.7.36"
+  testImplementation "ch.qos.logback:logback-classic:1.5.6"
```
Comment (rjrudin) [inline build.gradle:73]: Logback 1.4 requires Java 11 or higher, so stick with 1.3.x. Verified by running the tests locally on Java 8, and logback causes an error.

File: build.gradle [hunk 3]
```diff
@@ -63,13 +65,13 @@ dependencies {
...
+  testImplementation "org.slf4j:jcl-over-slf4j:2.0.13"
```
Comment (rjrudin) [inline build.gradle:74]: TIL that slf4j actually only requires Java 8, so this is a good upgrade to do in all of our projects.

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (BillFarber) [reply build.gradle:73]: sorry.

**Pull request merged**